src/jdk.jfr/share/classes/jdk/jfr/internal/EventInstrumentation.java

```diff
@@ -222,6 +246,30 @@ private void methodShouldCommit(CodeBuilder codeBuilder) {
             codeBuilder.invokevirtual(eventClassDesc, sd.methodName(), mdesc);
             codeBuilder.ifeq(fail);
         }
+        if (throttled) {
+            // long d =  eventConfiguration.throttle(this.duration);
+            // this.duration = d;
+            // if (d & MASK_THROTTLE_BIT == 0) {
+            //   goto fail;
+            // }
+            getEventConfiguration(codeBuilder);
+            codeBuilder.aload(0);
+            getfield(codeBuilder, eventClassDesc, ImplicitFields.FIELD_START_TIME);
```

Comment:
In native,we use the endTime for duration events? Is there a need to synchronize the two?

Reply:
The duration is added later, in throttle. There is no endTime field to read here.